### PR TITLE
Reduced to 2 the number of environments: development and production

### PR DESCRIPTION
**Pull request by cbeldacap**
### Removed
Different environments than:
- `dev` --> For local development (pointing to `localhost`)
- `prod` --> For production. It will point to custom URL when the Docker container is run passing some arguments like:

`docker run -it -d --name=mts-angular --build-arg APP_REST_PATH_ROOT=[custom-url] (...)`

This will override the production environment of the application setting up the [custom-url] instead.

### Changed
- `environment.prod.ts`
Check if the [custom-url] had been changed. If not, set it up one by default pointing to `localhost`.

### Added
- `Dockerfile.test`
Dockerfile to test the production environment. It also substitutes the ENV variables in `environment.prod.ts` by the ones inside of the container.

Steps:
1. `docker build -t [image-tag] --build-arg APP_REST_PATH_ROOT=http://[etc] --build-arg APP_REST_SERVICE_ROOT=http://[etc] .` (if no `--build-arg`(s) are set, the image will add some default values to `localhost`)
2. `docker run -it -d -p 4200:4200 [image-tag]`

### CODE REVIEW

File: angular/.angular-cli.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
         "prod": "environments/environment.prod.ts",
         "prodcompose": "environments/environment.prodcompose.ts",
         "node": "environments/environment.node.ts",
-        "net": "environments/environment.net.ts"
+        "net": "environments/environment.net.ts",
+        "openshift": "environments/environment.openshift.ts"
```
Comment (amarinso) [inline angular/.angular-cli.json]: why arent we using environment.prod.ts and instead making another new file? this configuration should be valid not only for docker but also for other docker deployments

File: angular/src/app/book-table/shared/book-table.service.spec.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { TestBed, inject } from '@angular/core/testing';
-import { HttpModule} from '@angular/http';
+import { HttpModule } from '@angular/http';
```
Comment (amarinso) [inline angular/src/app/book-table/shared/book-table.service.spec.ts]: doesn't this formatting changes be addressed on a separate PR to keep a clean history? do we have any politic about this?

File: angular/.angular-cli.json (hunk 1 above)
Comment (amarinso) [inline angular/.angular-cli.json]: I think that we should add also the same changes to the default development environment so we only work with the "standard" angular environments default(dev) and prod

of course projects can create their own environments but we leave that option for more exotic ones, not for the common case

also we should include a comment on the file explaining the logic behind the "extrange" code because it is not that intuitive

File: angular/src/app/book-table/shared/book-table.service.spec.ts (hunk 2 above)
Comment (cbeldacap) [inline angular/src/app/book-table/shared/book-table.service.spec.ts]: well, it makes sense. I'll do so from now on @amarinso 

### updated PR
